src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+            // anything (so that the property will fall back into the next value).
+            for (String property : propertyNames) {
+
+                if ((!environmentProperties.keySet().contains(property)) &&
+                        (!systemProperties.containsKey(property))) {
+
+                    ConfigFactory.setProperty(property, "/dev/null");
+                }
+            }
+
+            hasSetConfigFactoryVariableDefaults = true;
+        }
+    }
+
+    /**
+     * Get the configuration file name from the given arguments.
+     * Modifies the given arguments to remove both the configuration file specification string
+     * and the configuration file name from the args.
+     *
+     * NOTE: Does NOT validate that the resulting string is a valid configuration file.
+     *
+     * @param args Command-line arguments passed to this program.
+     * @param configFileOption The command-line option indicating that the config file is next
+     * @return The name of the configuration file for this program or {@code null}.
+     */
+    public static final String getConfigFilenameFromArgs( final ArrayList<String> args, final String configFileOption ) {
+
+        Utils.nonNull(args);
+        Utils.nonNull(configFileOption);
+
+        String configFileName = null;
+
+        for ( int i = 0 ; i < args.size() ; ++i ) {
+            if (args.get(i).compareTo(configFileOption) == 0) {
+
+                // Get rid of the command-line argument name:
+                args.remove(i);
+
+                if ( i < args.size() ) {
+
+                    // Get and remove the specified config file:
+                    configFileName = args.remove(i);
+                    break;
+                }
+                else {
+                    // Option was provided, but no file was specified.
+                    // We cannot work under these conditions:
+
+                    String message = "ERROR: Configuration file not given after config file option specified: " + configFileOption;
+                    System.err.println(message);
+                    throw new UserException.BadInput(message);
+                }
+            }
+        }
+
+        return configFileName;
+    }
+
+    /**
+     * Get the given key from a {@link GATKConfig} class.
+     * @param key The key in the {@link GATKConfig} to get.
+     * @param <T> The type of the key to obtain.
+     * @return The value for the specified {@code key}.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T getFromConfig(String key) {
+
+        Utils.nonNull(key);
+
+        // Get our configuration:
+        final GATKConfig config = ConfigCache.getOrCreate( GATKConfig.class );
+
+        try {
+            for (Method m : GATKConfig.class.getDeclaredMethods()) {
+                if (m.getName().equals(key)) {
+                    return (T) m.invoke(config);
+                }
+            }
+        }
+        catch (IllegalAccessException e) {
```

Comment:
`catch` should be on the same line as the `}` for the `try`. In this case you can also combine both exceptions in a single catch, since their bodies are the same. E.g.
```
} catch (IllegalAccessException | InvocationTargetException e) {
    logger.fatal(...);
}
```